### DISCUSSION TITLE
Disable using the GitHub mobile app as MFA

### DISCUSSION THREAD

**Discussion by sammcj** [community/community#30502] [Mobile] [answered]
It's really annoying that GitHub keeps trying to use the mobile app by default rather than my chosen MFA.

There needs to be a way to disable this for the mobile app.

<details>
<summary>ℹ️ Additional Debug Info</summary>
GitHub for iOS, Version 1.77.0 (76215101)
Device: iPhone 12 Pro (iOS 16.0)
</details>

**Comment by taki-on** [comment]
Hi @sammcj the issue is known to us, and we are currently working on better 2FA experiences regarding the user's authentication app setting. Please stay tuned!

**Comment by eliperkins** [comment, marked as answer]
Thanks for your feedback! Today, we've shipped [a new feature that allows for setting a preferred option for two-factor authentication](https://github.blog/changelog/2023-02-22-preferred-2fa-methods-and-settings-improvements/). This will allow you to set other two-factor authentication methods to be used before another option. [Go to Settings > Password and Authentication on GitHub.com](https://github.com/settings/security) in your browser to set a different preferred two-factor option.

<img width="772" alt="Screenshot 2023-02-22 at 4 46 23 PM" src="https://user-images.githubusercontent.com/1051453/220766920-fc0cacc4-5539-4afc-9940-83acbd17efee.png">

Additionally, [GitHub Mobile sessions can be revoked within Settings](https://github.com/settings/sessions?#mobile_sessions) as well:

<img width="793" alt="Screenshot 2023-02-22 at 4 49 54 PM" src="https://user-images.githubusercontent.com/1051453/220767378-9bdca298-994d-4cb8-8b7c-22724b16e2a1.png">

**Reply by liamriley1** [reply]
Is there a way I can completely disable this method all together? Whilst the "preferred method" option does mean that I don't have to deselect github mobile as the was to authenticate myself each time I log in,  I do not want it to be available as a 2FA method for accessing my account at all. In my opinion, it weakens the security of the 2FA method that I actually want (security key) by offering potential attackers an easier way to attack my account. Please stop forcing this weaker github mobile 2FA on people and allow the users to choose their own 2FA methods, whilst still using the mobile app as normal. At the moment, the only workaround to this is to not use the mobile app, which is much less convenient for many users.

**Reply by cuducos** [reply]
> Please stop forcing this weaker github mobile 2FA on people

That. Please.

**Reply by Norbiros** [reply]
@eliperkins any updates? Is there any way to completely disable MFA using Github App? I don't want to revoke sessions or change preferred method. For my normal 2FA I have pin & fingerprint protection, here if you have access to my unlocked phone you can do destructive things on my gh account

**Comment by SjoerdV** [comment]
I agree with @sammcj and @lriley2020 especially since:
- a dedicated authenticator app is preferred (not mixing content and security). I really dislike the Outlook app for that as well!
- the Github mobile app does not have a 'TouchId' (or other biometric / password) setting in the App, making it insecure in itself (at least Outlook has this covered)
- I don't like 'everything' apps, they are security and privacy nightmares

I will go as far as removing the App from all my devices and use browser only, if this is not fixed within a reasonable time frame. Thanks!

**Comment by DuneArchitect** [comment]
> it weakens the security of the 2FA method that I actually want (security key) by offering potential attackers an easier way to attack my account

Hi, I agree with all the above **allowing Github mobile and SMS as multi-factor options reduces the overall security of the account.**

My multi-factor app of choice forces authentication on opening.  The above two are already accessible if the phone has already been unlocked, such as handing it to someone to make a phone call, check directions, etc.

@eliperkins while the option to set a preferred method improves the user experience, there is a still a security bug here:  *Users need the option to disable mulit-factor methods as it improves the overall security of the account.*

**Comment by 0xDracarys** [comment]
I want to disable my mfa, i am unable to login

**Comment by jcstop** [comment]
How hard can it be to implement an option to not allow github app for authentication. How is it possible that it takes years for this obvious configuration to be added??
